src/subclass/mod.rs

```diff
@@ -55,143 +55,161 @@
+//!                     self.name.replace(name);
+//!                 },
+//!                 subclass::Property("animal", ..) => {
+//!                     let animal = value
+//!                         .get()
+//!                         .expect("type conformity checked by `Object::set_property`");
+//!                     self.animal.replace(animal.unwrap());
+//!                 },
+//!                 subclass::Property("flags", ..) => {
+//!                     let flags = value
+//!                         .get()
+//!                         .expect("type conformity checked by `Object::set_property`");
+//!                     self.flags.replace(flags.unwrap());
+//!                 },
+//!                 _ => unimplemented!(),
+//!             }
 //!         }
-//!     }
 //!
-//!     // Called whenever a property is retrieved from this instance. The id
-//!     // is the same as the index of the property in the PROPERTIES array.
-//!     fn get_property(&self, _obj: &glib::Object, id: usize) -> Result<glib::Value, ()> {
-//!         let prop = &PROPERTIES[id];
+//!         // Called whenever a property is retrieved from this instance. The id
+//!         // is the same as the index of the property in the PROPERTIES array.
+//!         fn get_property(&self, _obj: &glib::Object, id: usize) -> Result<glib::Value, ()> {
+//!             let prop = &PROPERTIES[id];
+//!
+//!             match *prop {
+//!                 subclass::Property("name", ..) => Ok(self.name.borrow().to_value()),
+//!                 subclass::Property("animal", ..) => Ok(self.animal.get().to_value()),
+//!                 subclass::Property("flags", ..) => Ok(self.flags.get().to_value()),
+//!                 _ => unimplemented!(),
+//!             }
+//!         }
+//!
+//!         // Called right after construction of the instance.
+//!         fn constructed(&self, obj: &glib::Object) {
+//!                 // Chain up to the parent type's implementation of this virtual
+//!             // method.
+//!             self.parent_constructed(obj);
 //!
-//!         match *prop {
-//!             subclass::Property("name", ..) => Ok(self.name.borrow().to_value()),
-//!             subclass::Property("animal", ..) => Ok(self.animal.get().to_value()),
-//!             subclass::Property("flags", ..) => Ok(self.flags.get().to_value()),
-//!             _ => unimplemented!(),
+//!             // And here we could do our own initialization.
 //!         }
 //!     }
+//! }
 //!
-//!     // Called right after construction of the instance.
-//!     fn constructed(&self, obj: &glib::Object) {
-//!         // Chain up to the parent type's implementation of this virtual
-//!         // method.
-//!         self.parent_constructed(obj);
+//! // Optionally, define a wrapper type to make it more ergonomic to use from Rust
+//! glib_wrapper! {
+//!     pub struct SimpleObject(ObjectSubclass<imp::SimpleObject, SimpleObjectClass>);
```

Comment:
```suggestion
//!     pub struct SimpleObject(ObjectSubclass<imp::SimpleObject, imp::SimpleObjectClass>);
```

shouldn't it be like that?

Reply:
No, because this is the name that is given to the wrapper (confusingly). The actual class struct is taken from `ObjectSubclass::Class`